atom-client/src/main/java/org/motechproject/atomclient/service/impl/FeedCache.java

```diff
@@ -0,0 +1,290 @@
+            }
+        }
+
+        return null;
+    }
+
+
+    /**
+     * Sends a MOTECH event for a feed entry
+     *
+     * @param entry
+     */
+    private void sendMessageForFeedEntry(String url, SyndEntry entry, String regex) {
+        Map<String, Object> parameters = new HashMap<>();
+        Map<String, Object> rawContent = new HashMap<>();
+        Map<String, Object> extractedContent = new HashMap<>();
+        parameters.put("url", url);
+        parameters.put("published_date", entry.getPublishedDate());
+        parameters.put("updated_date", entry.getUpdatedDate());
+        if (entry.getContents() != null) {
+            Integer index = 1;
+            for (SyndContent content : entry.getContents()) {
+                rawContent.put(index.toString(), content.getValue());
+                String extractedContentString = extractContent(content.getValue(), regex);
+                if (!StringUtils.isBlank(extractedContentString)) {
+                    extractedContent.put(index.toString(), extractedContentString);
+                }
+                index++;
+            }
+        } else {
+            LOGGER.warn("NULL content for entry {}", entry.getUri());
+        }
+        parameters.put("raw_content", rawContent);
+        parameters.put("extracted_content", extractedContent);
+        MotechEvent event = new MotechEvent(Constants.FEED_CHANGE_MESSAGE, parameters);
+        LOGGER.debug("sending message {}", event);
+        eventRelay.sendEventMessage(event);
+    }
+
+
+    /**
+     * Sends a MOTECH event for each new feed entry
+     *
+     * @param feed
+     */
+    private void sendMessagesForNewFeedData(String url, SyndFeed feed, String regex) {
+        for (SyndEntry entry : feed.getEntries()) {
+            sendMessageForFeedEntry(url, entry, regex);
+        }
+    }
+
+
+    private static boolean areEntriesDifferent(SyndEntry e1, SyndEntry e2) {
+        if (!ObjectUtils.equals(e1.getUpdatedDate(), e2.getUpdatedDate())) {
+            LOGGER.trace("different updated date");
+            return true;
+        }
+        if (!ObjectUtils.equals(e1.getPublishedDate(), e2.getPublishedDate())) {
+            LOGGER.trace("different published date");
+            return true;
+        }
+        if (!ObjectUtils.equals(e1.getUri(), e2.getUri())) {
+            LOGGER.trace("different uri");
+            return true;
+        }
+        if (!ObjectUtils.equals(e1.getContents(), e2.getContents())) {
+            LOGGER.trace("different contents");
+            return true;
+        }
+
+        return false;
+    }
+
+
+    /**
+     * Sends a MOTECH event for each changed feed entry and returns true if changes were detected
+     *
+     * @param cachedFeed
+     * @param fetchedFeed
+     * @return true if any changes were detected between the cached entry and the fetched entry
+     */
+    private boolean sendMessagesForChangedEntries(String url, SyndFeed cachedFeed, SyndFeed fetchedFeed, String regex) {
```

Comment:
Seems like a good method to add loggers in - whether the entries were in cache or not and if there were any changes